lib/ansible/module_utils/basic.py

```diff
@@ -1211,7 +1211,7 @@ def _check_required_arguments(self):
         missing = []
         for (k,v) in self.argument_spec.items():
             required = v.get('required', False)
-            if required and k not in self.params:
+            if required and self.params.get(k) is None:
```

Comment:
This should prevent a missing argument from skipping the required argument check